modules/accredited_representative_portal/app/controllers/accredited_representative_portal/v0/decisions_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module AccreditedRepresentativePortal
+    module V0
+        class DecisionsController < ApplicationController
+            before_action :set_power_of_attorney_request, only: [:show, :create]
+
+            def create
+                if @power_of_attorney_request.resolution.present?
```

Comment:
This logic should go in our model or service layer, I'd highly favor model layer for now. Actually a uniqueness validation is just this: [guide](https://guides.rubyonrails.org/active_record_validations.html#uniqueness).

Reply:
so if I take this guard clause at the top out, I was getting a duplicate index error so just put it in a begin rescue

` begin
          @power_of_attorney_request.create_resolution!( 
            resolving: PowerOfAttorneyRequestDecision.new(creation_params), 
            reason:
          )
          resolution = @power_of_attorney_request.resolution
          if resolution.persisted?
            decision = resolution.resolving
            render json: decision, status: :ok
          else 
            render json: { error: 'Failed to create resolution' }, status: :unprocessable_entity
          end
        rescue => e
          Rails.logger.error("ARP: Unexpected error occurred for power of attorney request #{@power_of_attorney_request.id} - #{e.message}")
          render json: { error: 'Failed to create resolution' }, status: :unprocessable_entity
        end`
        
  Then it goes back to generic error so will probably add in a specific rescue for that error for duplicate to return better info

Reply:
weird that code is normally spaced for me when pasting it in but super ugly for you. Here's a screenshot
<img width="1001" alt="Screenshot 2024-12-23 at 2 33 16 PM" src="https://github.com/user-attachments/assets/e5bf93c6-2653-4340-970a-1735d82f228b" />